lib/puppet/transaction/persistence.rb

```diff
@@ -0,0 +1,81 @@
+require 'yaml'
+require 'puppet/util/yaml'
+
+# A persistence store implementation for storing information between
+# transaction runs for the purposes of information inference (such
+# as calculating corrective_change).
+# @api private
+class Puppet::Transaction::Persistence
```

Comment:
Ah, so you've already been chatting with those guys on this - great.  That reasoning all makes sense.

Sorry I'm a bit late to the party then - carry on!